src/test/java/de/unijena/cheminf/fragment/fingerprint/ExampleUsageTest.java

```diff
@@ -0,0 +1,214 @@
+            System.out.println(tmpFingerprinter.getBitDefinition(i) + ": " + tmpMyPartyPeopleFP.get(i));
+        }
+        /*
+         * Output:
+         * Hannah: true
+         * Sam: true
+         * John: false
+         * Hugo: true
+         * Tim: false
+         */
+    }
+    //
+    /**
+     * The intended use case of the fragment fingerprinter functionality is to encode the presence and absence of
+     * substructures in a given molecule that result from a molecular fragmentation study, i.e. the algorithmic
+     * extraction of specific substructures from input molecules. These substructures are automatically extracted and
+     * can be represented by different string-based molecular structure encodings, like SMILES or InChI. Other
+     * key-based substructure fingerprint functionalities require SMARTS strings as inputs and are therefore not
+     * as ubiquitously applicable as the fragment fingerprint for this purpose.
+     *
+     * In the following, a molecular structure data set is imported that contains 100 natural products with a
+     * naphthalene substructure taken from the COCONUT natural products database. These are fragmented using the CDK
+     * ExhaustiveFragmenter functionality that breaks single non-ring bonds in input molecules to generate fragments.
+     * The resulting fragments are collected together with their fraquencies as unique SMILES representations.
+     * Fragments that occur more than two times are then used to initialise the fragment fingerprinter. At the end,
+     * the "naphthalene-derivatives exhaustive fragmenter fingerprint" is generated for 3-hydroxy-2-naphthoic acid.
+     */
+    @Test
+    public void chemicalExampleUsageTest() throws Exception {
+        //Note: make it "path\\to\\coconut_naphthalene_substructure_search_result.sdf" in the example code (I don't want to use the import from test resources here)
+        IteratingSDFReader tmpSDFReader = new IteratingSDFReader(new FileInputStream(
+                "D:\\Project_Fragment_Fingerprints\\FragmentFingerprints_git\\src\\test\\resources\\de\\unijena\\cheminf\\" +
+                        "fragment\\fingerprint\\coconut_naphthalene_substructure_search_result.sdf"),
+                SilentChemObjectBuilder.getInstance());
+        //This fragmentation scheme simply breaks single non-ring bonds.
+        ExhaustiveFragmenter tmpFragmenter = new ExhaustiveFragmenter();
+        //Default would be 6 which is too high for the short side chains in the input molecules
+        tmpFragmenter.setMinimumFragmentSize(1);
+        SmilesGenerator tmpSmiGen = new SmilesGenerator(SmiFlavor.Unique);
+        HashMap<String, Integer> tmpFrequenciesMap = new HashMap<>(50, 0.75f);
+        while (tmpSDFReader.hasNext()) {
+            IAtomContainer tmpMolecule = tmpSDFReader.next();
+            tmpFragmenter.generateFragments(tmpMolecule);
+            IAtomContainer[] tmpFragments = tmpFragmenter.getFragmentsAsContainers();
```

Comment:
Why the usage of getFragmentsAsContainers() and not getFragments(), which returns them already as unique SMILES?
At the end, there is no big difference. I ask out of interest.

Reply:
I like to have full control via my own SmilesGenerator instance here. And I think it is also important in this context to show this explicitly. Not every fragmentation functionality will have this kind of convenience function. In addition, every other string-based representation could be used instead of SMILES.

Reply:
Perhaps a brief comment will make this even clearer.

Reply:
Added comment in last commit

Reply:
I find the example very illustrative and a very appropriate chemical example.

and i can gladly add real tests